akka-javasdk-testkit/src/main/java/akka/javasdk/testkit/TestModelProvider.java

```diff
@@ -26,16 +29,27 @@
  */
 public class TestModelProvider implements ModelProvider.Custom {
 
-  private List<Pair<Predicate<String>, String>> responsePredicates = new ArrayList<>();
+
+  public record ToolCallRequest(String name, String arguments) {
+  }
+
+  public record AiAnswer(String message, List<ToolCallRequest> requests) {
```

Comment:
I had to extend this with AiAnswer and ToolCallRequest to be able to support tool call. 

It turns out that in order to really simulate how a model will react, users will need to have a better understanding of it. For example, will the model return one single ToolCallRequest or many? In which situation will the model ask for parallel tool call? 

This will make the mocking harder.